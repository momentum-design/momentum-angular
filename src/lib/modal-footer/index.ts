export * from './modal-footer.module';
export * from './modal-footer.component';
