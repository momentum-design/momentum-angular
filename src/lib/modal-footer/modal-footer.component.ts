import { Component, OnInit, Input, HostBinding } from '@angular/core';

@Component({
  selector: 'cui-modal-footer',
  template: `
    <ng-content></ng-content>
  `,
  styles: [],
})
export class ModalFooterComponent implements OnInit {
  /** @option css class names | '' | '' */
  @Input() public class: string = '';

  @HostBinding('class') get className(): string {
    return (
      'cui-modal__footer' + `${(this.class && ` ${this.class}`) || ''}` + ``
    );
  }

  constructor() {}

  ngOnInit() {}
}

/**
 * @component modal-footer
 * @section class
 * @angular
 *
<cui-modal>
  <cui-modal-footer class='large'></cui-modal-footer>
</cui-modal>
 */

/**
 * @component modal-footer
 * @section ng-content
 * @angular
 *
<cui-modal>
  <cui-modal-footer class='large'>
    <a>close</a>
  </cui-modal-footer>
</cui-modal>
 */
