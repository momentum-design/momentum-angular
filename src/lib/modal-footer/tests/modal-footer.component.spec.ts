import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ModalFooterComponent } from '../modal-footer.component';

describe('ModalFooterComponent', () => {
  let component: ModalFooterComponent;
  let fixture: ComponentFixture<ModalFooterComponent>;
  let testElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ModalFooterComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ModalFooterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render classname', () => {
    component.class = 'textAddOn';
    fixture.detectChanges();
    testElement = fixture.nativeElement;
    expect(testElement.className).toContain('textAddOn');
  });
});
