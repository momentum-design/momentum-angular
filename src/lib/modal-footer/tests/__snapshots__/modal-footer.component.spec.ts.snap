// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooterComponent should create and match snapshot 1`] = `
<cui-modal-footer
  class=""
>
  
</cui-modal-footer>
`;
